.github/workflows/release.yml

```diff
@@ -4,46 +4,197 @@ on:
+
+    outputs:
+      checksum: ${{ steps.publish.outputs.checksum }}
+      download: ${{ steps.publish.outputs.download }}
+      filename: ${{ steps.publish.outputs.filename }}
+      version: ${{ steps.publish.outputs.version }}
+
+
+  guide:
+    needs: source
+    environment: website
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/cockpit-project/unit-tests
+      options: --user root
+    steps:
+      - name: Checkout website repository
+        uses: actions/checkout@v3
+        with:
+          path: website
+          repository: cockpit-project/cockpit-project.github.io
+          ssh-key: ${{ secrets.DEPLOY_KEY }}
+
+      - name: Download source release
+        run: curl -L -o '${{ needs.source.outputs.filename }}' '${{ needs.source.outputs.download }}'
+
+      - name: Verify checksum
+        run: echo '${{ needs.source.outputs.checksum }} ${{ needs.source.outputs.filename }}' | sha256sum -c
+
+      - name: Build guide
         run: |
-          # override GitHub's bind mount from host, we don't want anything from there and it interferes with ssh
-          export HOME=$(getent passwd $(id -u) | cut -f6 -d:)
+          mkdir source build
+          tar --directory source --extract --strip-components=1 --file '${{ needs.source.outputs.filename }}'
+          (
+              cd build
+              ../source/configure
+              make doc/guide/html/index.html
+          )
 
-          git config user.name "GitHub Workflow"
-          git config user.email "cockpituous@cockpit-project.org"
+      - name: Update the website
+        run: |
+          rm -rf website/guide/latest
+          mv -Tv build/doc/guide/html website/guide/latest
 
-          echo '${{ secrets.GITHUB_TOKEN }}' > ~/.config/github-token
+          # Add frontmatter for Jekyll
+          find website/guide/latest -name '*.html' -exec sed -i '
+          1i\
+          ---\
+          layout: guide\
+          ---' '{}' ';'
 
-          # deploy key for updating the website repo, via tools/release-guide
-          eval $(ssh-agent)
-          ssh-add - <<< '${{ secrets.WEBSITE_DEPLOY_KEY }}'
+          git config --global user.name "GitHub Workflow"
+          git config --global user.email "cockpituous@cockpit-project.org"
+
+          cd website
+          git add guide/
+          git commit --message='Update guide to version ${{ needs.source.outputs.version }}'
+          git show --stat
+          git push origin main
+
+
+  flathub:
+    needs: source
+    environment: flathub
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source repository
+        uses: actions/checkout@v3
+        with:
+          path: src
 
-      - name: Run cockpituous
+      - name: Checkout flathub repository
+        uses: actions/checkout@v3
+        with:
+          path: flathub
+          repository: flathub/org.cockpit_project.CockpitClient
```

Comment:
We can rely on actions/checkout providing this "flathub" alias for us? Still feels easier to understand and reproduce to just spell it out here.

Reply:
I don't understand the question...

Reply:
Err, nevermind; I thought "flathub" was just some alias for git.flathub.org/blah/..., but obviously it's just a github org.